### PR TITLE
Sequelize's death

### LINKED ISSUE #69: Replace Sequelize with Knex

### PR DESCRIPTION
**Pull request by abrden**
Closes #69 :tada:

### CODE REVIEW

File: src/database/knexfile.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    client: 'postgresql',
+    connection: process.env.DATABASE_URL,
+    migrations: {
+      directory: __dirname + "/migrations",
+      tableName: 'knex_migrations'
+    }
+  },
+  test: {
+    client: 'postgresql',
+    connection: process.env.TEST_DATABASE_URL || 'postgres://postgres:postgres@localhost:5432/tallerify_test2',
```
Comment (manuporto) [inline src/database/knexfile.js]: Switch back to _'postgres://postgres:postgres@localhost:5432/tallerify_test'_

File: src/database/migrations/migrations.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const logger = require('../../utils/logger');
+const tables = require('../tableNames');
+
+exports.up = (knex, Promise) => {
+	return Promise.all([
```
Comment (manuporto) [inline src/database/migrations/migrations.js]: Do not mix indentations

File: src/handlers/db/generalHandler.js [hunk 3]
```diff
@@ -1,52 +1,37 @@
...
   });
 };
 
 const createNewEntry = (tableName, entry) => {
   logger.info('Creating entry');
-  return tables[tableName].create(entry);
+  return db(tableName).insert(entry).returning('*');
 };
 
-function updateEntry(entry, newEntry) {
```
Comment (manuporto) [inline src/handlers/db/generalHandler.js]: We should decide whether to use arrow functions or functions, but not mix between the two.

**Comment by abrden** [comment]
KB TINCHO TE GANAMOS EN COVERAGE :boom: @mastanca @saantiaguilera

File: src/handlers/db/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const general = require('./generalHandler');
+const artist = require('./artistHandler');
+const track = require('./trackHandler');
+
+const dbHandler = {
+  general,
+  artist,
+	track,
```
Comment (manuporto) [inline src/handlers/db/index.js]: Do not mix indentations

**Comment by manuporto** [comment]
Doing review

File: src/handlers/response.js [hunk 5]
```diff
@@ -11,21 +10,27 @@ const internalServerError = (reason, response) => {
   return response.status(500).json({ code: 500, message });
 };
 
-function validateJson(body, schema, callback) {
```
Comment (manuporto) [inline src/handlers/response.js]: Do not mix between arrow functions and functions

File: src/routes/album.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 const logger = require('../utils/logger');
-const models = require('../models/index');
 
 const postAlbum = (req, res) => {
+  /*
```
Comment (manuporto) [inline src/routes/album.js]: Delete commented code

File: src/routes/index.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ const user = require('./user');
 const token = require('./token');
 const artist = require('./artist');
 const track = require('./track');
-const album = require('./album');
+// const album = require('./album');
```
Comment (manuporto) [inline src/routes/index.js]: Delete commented code

File: src/routes/track.js [hunk 8]
```diff
@@ -1,15 +1,46 @@
...
+const getTracks = (req, res) => {
+  db.general.findAllEntries(tables.tracks)
+    .then(tracks => respond.succesfulTracksFetch(tracks, res))
+    .catch(error => respond.internalServerError(error, res));
+};
+
+const newTrack = (req, res) => {
+  respond.validateRequestBody(req.body, trackExpectedBodySchema)
+  .then(() => {
+    db.track.insertTrack(req.body) // FIXME validate body
```
Comment (manuporto) [inline src/routes/track.js]: `FIXME` no longer valid

File: src/test/routes/artist.spec.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  });
+
+  afterEach((done) => {
+    db.migrate.rollback()
+    .then(() => done());
+  });
+
+	describe('/GET artists', () => {
+
+		it('should return status code 200', (done) => {
```
Comment (manuporto) [inline src/test/routes/artist.spec.js]: Fix indentation

File: src/handlers/db/generalHandler.js (hunk 3 above)
Comment (abrden) [inline src/handlers/db/generalHandler.js]: arrow

**Comment by saantiaguilera** [comment]
@abrden @manuporto 90%...
image

**Pull request merged**